CONTRIBUTING.md

```diff
@@ -0,0 +1,126 @@
+# CONTRIBUTING
+
+Thank you for considering contributing to the Outrigger CLI!
+
+## Quality Contributions
+
+* Make sure your branch will compile.
+* Make sure your branch passes our static analysis checks.
+* Make sure your branch conforms with go fmt standards.
+* Manually test your changes.
+
+## User Interactions
+
+One of the key goals of this project is to promote a positive developer
+experience. Every interaction should be thought of with the following points:
+
+* Are you providing the user with enough context about what's being asked or being done?
+* Does the user expect to wait? Might the user think the tool stalled?
+* Is there black box business happening that could be made more transparent?
+
+We have a slightly complex logging API to support addressing these concerns.
+(See ./util/logging.go)
+
+Here are a few conventions:
+
+* **Starting a task that could take more than 5 seconds:**
+  * `cmd.out.Spin("Preparing the sauce")`
+* **Use the correct method to log operational results: (Pick one)**
+  * `cmd.out.Info("Sauce is Ready.")`
+  * `cmd.out.Warning("Sauce is burnt on the bottom.")`
+  * `cmd.out.Error("Discard this sauce and try again.")`
+* **Going to send some contextual notes to the user**:
```

Comment:
Is this example one where you always want to send a note to the user (or is the info/verbose a distinction on whether it's always versus only if --verbose is set)? If it's an indicator that the spinner needs to be disabled in those cases should we put re-enabling it after sending the message?

I don't see any indications of using NoSpin in the code so far other than in a Failure message so that makes me think perhaps I'm not interpreting how/when it needs to be used in order to use it correctly.

Reply:
It could be that NoSpin needs to be added somewhere. If you do some regular logs when the Spinner is active, it suspends the spinner on line 1, outputs the log on line 2, and resumes the spinner on line 3. If you add NoSpin before logging, it does not resume on line 3.